nft-client/src/main.rs

```diff
@@ -0,0 +1,635 @@
+                .context("Failed creating PostgreSQL 'notify_insert' function")?;
+
+            db_client
+                .batch_execute(
+                    "CREATE TRIGGER notify_on_insert_into_cti AFTER INSERT ON cti FOR EACH ROW \
+                     EXECUTE PROCEDURE pg_temp.notify_insert();
+                         LISTEN contract_updates;",
+                )
+                .await
+                .context("Failed creating PostgreSQL trigger")?;
+
+            // Await notifications, find and parse the relevant transaction summaries
+            println!("Started tracing contract events:");
+            while let Some(notification) = notifications.next().await {
+                if notification.channel() != "contract_updates" {
+                    continue;
+                }
+                let summary_index: i64 = notification.payload().parse()?;
+
+                let row = db_client
+                    .query_one("SELECT summary FROM summaries WHERE id = $1", &[
+                        &summary_index,
+                    ])
+                    .await?;
+                let summary = serde_json::from_value::<DatabaseSummaryEntry>(row.get(0))?;
+                let events = collect_summary_contract_events(summary, contract.0);
+                for event in events {
+                    let mut cursor = concordium_contracts_common::Cursor::new(event.as_ref());
+                    let event = cts1::Event::deserial(&mut cursor)
+                        .map_err(|_| anyhow!("Failed parsing event"))?;
+                    println!("{}", event);
+                }
+            }
+        }
+        Command::Mint {
+            sender,
+            token_ids,
+            contract,
+        } => {
+            // The GRPC client for Concordium node.
+            let mut grpc_client = Client::connect(app.endpoint, "rpcadmin".to_string())
+                .await
+                .context("Failed to connect to Node GRPC")?;
+
+            let account_data: AccountData = serde_json::from_str(
+                &std::fs::read_to_string(sender).context("Could not read the keys file.")?,
+            )
+            .context("Could not parse the accounts file.")?;
+
+            let owner = convert_account_address(&account_data.address);
+            let mint_parameter = cts1::MintParams {
+                owner:     concordium_contracts_common::Address::Account(owner),
+                token_ids: token_ids.clone(),
+            };
+            let bytes = concordium_contracts_common::to_bytes(&mint_parameter);
+
+            let update_payload = transactions::UpdateContractPayload {
+                amount:       common::types::Amount::from(0),
+                address:      contract.0,
+                receive_name: smart_contracts::ReceiveName::try_from(format!(
+                    "{}.mint",
+                    NFT_CONTRACT_NAME
+                ))
+                .map_err(|e| anyhow!("Failed to parse receive name {}", e))?,
```

Comment:
Why not `.context`?

Reply:
NewReceiveNameError does not implement Error